### PR TITLE
Update to latest rust

### PR DESCRIPTION
**Pull request by Potpourri**
Add necessary lifetime bound and remove unnecessary cast.

### CODE REVIEW

File: src/device/mapping.rs [hunk 1]
```diff
@@ -122,22 +130,24 @@ impl<'a, T: Copy, R: Resources, F: Factory<R>> Drop for RW<'a, T, R, F> where
 /// A service trait with methods for mapping already implemented.
 /// To be used by device back ends.
 #[allow(missing_docs)]
-pub trait Builder<R: Resources> {
+pub trait Builder<'a, R: Resources> {
     type RawMapping: Raw;
 
-    fn map_readable<T: Copy>(&mut self, Self::RawMapping, usize)
+    fn map_readable<T: Copy>(&'a mut self, Self::RawMapping, usize)
                     -> Readable<T, R, Self>;
```
Comment (kvark) [inline src/device/mapping.rs]: shouldn't this be using `'a` as well?

**Comment by kvark** [comment]
Thanks @Potpourri ! Just one small thing to address.

File: src/device/mapping.rs (hunk 1 above)
Comment (Potpourri) [inline src/device/mapping.rs]: You mean `Readable<'a, T, R, Self>`? If yes then shouldn't, rust inference lifetime from self.

Comment (kvark) [inline src/device/mapping.rs]: Right. But. I don't know what rust guidelines tell, but I'd expect the lifetimes to either be fully specified, or fully inferred. Doing it half-way seems to be a bit confusing.

Anyway, not a blocked by any means. Merging this.

**Pull request merged**